File: TabCommon/Library/src/main/java/com/nextzy/library/base/mvvm/layer3Repository/network/base/DefaultClient.java
package com.nextzy.library.base.mvvm.layer3Repository.network.base;

import java.util.List;

import okhttp3.Interceptor;
import okhttp3.OkHttpClient;

/**
 * Created by「 The Khaeng 」on 29 Sep 2017 :)
 */

public class DefaultClient extends BaseClient{
    private static DefaultClient instance;

    public static DefaultClient getInstance() {
        if (instance == null) {
            instance = new DefaultClient();
        }
        return instance;
    }

    @Override
    public void configBuilder( OkHttpClient.Builder builder ){
        super.configBuilder( builder );
    }

    @Override
    public List<Interceptor> setupInterceptorList(){
        return super.setupInterceptorList();
    }
}

File: TabCommon/Library/src/main/java/com/nextzy/library/base/mvvm/layer3Repository/network/base/DefaultClient.kt
package com.nextzy.library.base.mvvm.layer3Repository.network.base

import okhttp3.Interceptor
import okhttp3.OkHttpClient

/**
 * Created by「 The Khaeng 」on 29 Sep 2017 :)
 */

class DefaultClient : BaseClient() {

    companion object {
        lateinit var instance: DefaultClient
            private set
    }

    init {
        instance = DefaultClient()
    }

    override
    fun configBuilder(builder: OkHttpClient.Builder) {
        super.configBuilder(builder)
    }

    override
    fun setupInterceptorList(): List<Interceptor> {
        return super.setupInterceptorList()
    }

}
